SPECS/openvswitch/openvswitch.spec

```diff
@@ -79,216 +90,275 @@ Requires:       %{name} = %{version}-%{release}
-for service in openvswitch ovsdb-server ovs-vswitchd ovn-controller ovn-controller-vtep ovn-northd; do
-	install -p -D -m 0644 rhel/usr_lib_systemd_system_${service}.service %{buildroot}/%{_unitdir}/${service}.service
-done
+install -p -D -m 0755 \
+        rhel/usr_share_openvswitch_scripts_ovs-systemd-reload \
+        $RPM_BUILD_ROOT%{_datadir}/openvswitch/scripts/ovs-systemd-reload
 
-mkdir -p %{buildroot}/%{_sysconfdir}/openvswitch
-install -p -D -m 0644 rhel/etc_openvswitch_default.conf %{buildroot}/%{_sysconfdir}/openvswitch/default.conf
-sed -i '/OVS_USER_ID=.*/c\OVS_USER_ID=' %{buildroot}/%{_sysconfdir}/openvswitch/default.conf
+# remove unpackaged files
+rm -f $RPM_BUILD_ROOT%{_bindir}/ovs-parse-backtrace
 
 %check
-make -k check |& tee %{_specdir}/%{name}-check-log || %{nocheck}
+touch resolv.conf
+export OVS_RESOLV_CONF=$(pwd)/resolv.conf
+make -k check TESTSUITEFLAGS='%{_smp_mflags}' RECHECK=yes |& tee %{_specdir}/%{name}-check-log || %{nocheck}
+
+%pre selinux-policy
+%selinux_relabel_pre -s targeted
 
 %preun
 %systemd_preun %{name}.service
 
-%preun -n ovn-central
-%systemd_preun ovn-northd.service
+%pre
+getent group openvswitch >/dev/null || groupadd -r openvswitch
+getent passwd openvswitch >/dev/null || \
+    useradd -r -g openvswitch -d / -s /sbin/nologin \
+    -c "Open vSwitch Daemons" openvswitch
 
-%preun -n ovn-host
-%systemd_preun ovn-controller.service
+getent group hugetlbfs >/dev/null || groupadd -r hugetlbfs
+usermod -a -G hugetlbfs openvswitch
 
-%preun -n ovn-controller-vtep
-%systemd_preun ovn-controller-vtep.service
+exit 0
 
 %post
+if [ $1 -eq 1 ]; then
+    %define gname hugetlbfs
+    sed -i \
+        's@^#OVS_USER_ID="openvswitch:openvswitch"@OVS_USER_ID="openvswitch:%{gname}"@'\
+        %{_sysconfdir}/sysconfig/openvswitch
+    sed -i 's:\(.*su\).*:\1 openvswitch %{gname}:' %{_sysconfdir}/logrotate.d/openvswitch
+
+    # In the case of upgrade, this is not needed
+    chown -R openvswitch:openvswitch %{_sysconfdir}/openvswitch
+    chown -R openvswitch:%{gname} %{_localstatedir}/log/openvswitch
+fi
+
+# This may not enable openvswitch service or do daemon-reload.
 %systemd_post %{name}.service
 
-%post -n ovn-central
-%systemd_post ovn-northd.service
-
-%post -n ovn-host
-%systemd_post ovn-controller.service
-
-%post -n ovn-controller-vtep
-%systemd_post ovn-controller-vtep.service
+%post selinux-policy
+%selinux_modules_install -s targeted %{_datadir}/selinux/packages/%{name}/openvswitch-custom.pp
 
 %postun
 %systemd_postun %{name}.service
 
-%postun -n ovn-central
-%systemd_postun ovn-northd.service
+%postun selinux-policy
+if [ $1 -eq 0 ] ; then
+  %selinux_modules_uninstall -s targeted openvswitch-custom
+fi
 
-%postun -n ovn-host
-%systemd_postun ovn-controller.service
+%posttrans selinux-policy
+%selinux_relabel_post -s targeted
 
-%postun -n ovn-controller-vtep
-%systemd_postun ovn-controller-vtep.service
-
-%files
+%files selinux-policy
 %defattr(-,root,root)
-%license LICENSE
```

Comment:
Now we're missing the `%license` macro in the new packages. We should add it back.